src/main/java/seedu/address/logic/parser/DeleteTaskCommandParser.java

```diff
@@ -0,0 +1,96 @@
+     * Parses the given {@code String} of arguments in the context of the DeleteTaskCommand and returns a
+     * DeleteTaskCommand object for execution.
+     *
+     * @param args String to parse.
+     * @return DeleteTaskCommand to execute.
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public DeleteTaskCommand parse(String args) throws ParseException {
+        DeleteTaskCommand deleteTaskCommand;
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_ID, PREFIX_INDEX, PREFIX_MODULE_CODE, PREFIX_TASK_NAME);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_ID) // supplied neither index nor id nor module code nor task name
+                && !arePrefixesPresent(argMultimap, PREFIX_INDEX)
+                && !arePrefixesPresent(argMultimap, PREFIX_MODULE_CODE)
+                && !arePrefixesPresent(argMultimap, PREFIX_TASK_NAME)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTaskCommand.MESSAGE_USAGE));
+        } else if (arePrefixesPresent(argMultimap, PREFIX_ID) // supplied either index or id
+                || arePrefixesPresent(argMultimap, PREFIX_INDEX)) {
+            deleteTaskCommand = indexOrStudentIdGiven(argMultimap);
+        } else {
+            deleteTaskCommand = moduleCodeOrTaskNameGiven(argMultimap);
+        }
+        return deleteTaskCommand;
+    }
+
+    @SuppressWarnings("OptionalGetWithoutIsPresent") // This is checked using our custom function (arePrefixesPresent).
+    private static DeleteTaskCommand indexOrStudentIdGiven(ArgumentMultimap argMultimap) throws ParseException {
+
+        if (arePrefixesPresent(argMultimap, PREFIX_ID) // supplied no index
+                && !arePrefixesPresent(argMultimap, PREFIX_INDEX)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTaskCommand.MESSAGE_USAGE));
+        } else if (!arePrefixesPresent(argMultimap, PREFIX_ID) // supplied no id
+                && arePrefixesPresent(argMultimap, PREFIX_INDEX)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTaskCommand.MESSAGE_USAGE));
+        } else {
+            try {
+                StudentId studentId = ParserUtil.parseStudentId(argMultimap.getValue(PREFIX_ID).get());
+                Index index = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_INDEX).get());
+                return new DeleteTaskCommand(studentId, index);
+            } catch (ParseException pe) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTaskCommand.MESSAGE_USAGE), pe);
+            }
+        }
+    }
+
+    @SuppressWarnings("OptionalGetWithoutIsPresent") // This is checked using our custom function (arePrefixesPresent).
+    private static DeleteTaskCommand moduleCodeOrTaskNameGiven(ArgumentMultimap argMultimap) throws ParseException {
```

Comment:
Similar to the above function, would a name like `moduleCodeAndTaskNameGiven` be more accurate for this function?

Reply:
The function was given the name,  `indexOrStudentIdGiven` , to reflect that `moduleCode` or `taskName` parameter is given, and the function is to check whether both the arguments are present, and only if both arguments are present it can implement that function. If one of the parameters is missing, we would have to raise on error. Hence, the 'or' is to reflect on this ambiguity. Please let me know what you think...